src/Sepes.Infrastructure/Service/UserService .cs

```diff
@@ -77,10 +80,48 @@ public bool CurrentUserIsDatasetAdmin()
             return _principalService.GetPrincipal().IsInRole(AppRoles.DatasetAdmin);
         }
 
-        UserDto MapToDtoAndAttachPrincipal(User user)
+        public  async Task<UserPermissionDto> GetUserPermissionsAsync()
+        {
+            var userFromDb = await GetCurrentUserFromDbAsync();
+
+            var result = new UserPermissionDto();
+            result.Admin = userFromDb.Admin;
+            result.Sponsor = userFromDb.Sponsor;
+            result.DatasetAdmin = userFromDb.DatasetAdmin;
+
+            result.CanCreateStudy = userFromDb.Admin || userFromDb.Sponsor;
+            result.CanAdministerDatasets = userFromDb.Admin || userFromDb.DatasetAdmin;
+
+            return result;
+        }
+
+        UserDto MapToDtoAndPersistRelevantProperties(User user)
         {
-            var mapped = _mapper.Map<UserDto>(user);        
-            return mapped;
+            var userDto = _mapper.Map<UserDto>(user);
+
+            if(userDto.ObjectId != _cachedUser.ObjectId)
```

Comment:
Where does _cachedUser come from?

Reply:
It's only cached in memory within the same request. (The user service only lives one request)
Since i know it's being used multiple times within the same request, it makes sense to cache it, especially to prevent multiple database rountrips for the same user info.